src/legacy/ui/public/vis/editors/default/components/sidebar/sidebar.tsx

```diff
@@ -0,0 +1,220 @@
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { useMemo, useState, useCallback, KeyboardEventHandler, useEffect } from 'react';
+import { get, isEqual } from 'lodash';
+import { i18n } from '@kbn/i18n';
+import { keyCodes, EuiButtonIcon, EuiFlexGroup, EuiFlexItem, EuiTitle } from '@elastic/eui';
+
+import { Vis } from 'ui/vis';
+import { PersistedState } from 'ui/persisted_state';
+import { DefaultEditorNavBar, OptionTab } from './navbar';
+import { DefaultEditorControls } from './controls';
+import { setStateParamValue, useEditorReducer, useEditorFormState } from './state';
+import { AggGroupNames } from '../../agg_groups';
+import { DefaultEditorAggCommonProps } from '../agg_common_props';
+
+interface DefaultEditorSideBarProps {
+  isCollapsed: boolean;
+  onClickCollapse: () => void;
+  optionTabs: OptionTab[];
+  uiState: PersistedState;
+  vis: Vis;
+}
+
+function DefaultEditorSideBar({
+  isCollapsed,
+  onClickCollapse,
+  optionTabs,
+  uiState,
+  vis,
+}: DefaultEditorSideBarProps) {
+  const [selectedTab, setSelectedTab] = useState(optionTabs[0].name);
+  const [isDirty, setDirty] = useState(false);
+  const [state, dispatch] = useEditorReducer(vis);
+  const { formState, setTouched, setValidity, resetValidity } = useEditorFormState();
+
+  const responseAggs = useMemo(() => state.aggs.getResponseAggs(), [state.aggs]);
+  const metricAggs = useMemo(
+    () => responseAggs.filter(agg => get(agg, 'schema.group') === AggGroupNames.Metrics),
+    [responseAggs]
+  );
+  const hasHistogramAgg = useMemo(() => responseAggs.some(agg => agg.type.name === 'histogram'), [
+    responseAggs,
+  ]);
+
+  const setStateValidity = useCallback(
+    (value: boolean) => {
+      setValidity('visOptions', value);
+    },
+    [setValidity]
+  );
+
+  const setStateValue: DefaultEditorAggCommonProps['setStateParamValue'] = useCallback(
+    (paramName, value) => {
+      const shouldUpdate = !isEqual(state.params[paramName], value);
+
+      if (shouldUpdate) {
+        dispatch(setStateParamValue(paramName, value));
+      }
+    },
+    [dispatch, state.params]
+  );
+
+  const applyChanges = useCallback(() => {
+    if (formState.invalid || !isDirty) {
+      setTouched(true);
+
+      return;
+    }
+
+    vis.setCurrentState(state);
+    vis.updateState();
+    vis.emit('dirtyStateChange', {
+      isDirty: false,
+    });
+    setDirty(false);
```

Comment:
Since this method is also invoked in `dirtyStateChange` event handler (line 110), I guess this call is redundant here.

Reply:
Thanks for the notice, done!